src/trivia/util.h

```diff
@@ -429,6 +429,27 @@ alloc_failure(const char *filename, int line, size_t size)
 #  define ALWAYS_INLINE inline
 #endif
 
+/**
+ * Specifies that an object or member declaration with type array of char,
+ * signed char, or unsigned char, or pointer to such a type is intended to
+ * store character arrays that do not necessarily contain a terminating NULL.
+ *
+ * In particular prevents warnings from compiler when initializing array
+ * of chars by string literal truncates terminating NULL.
+ *
+ * Example:
+ *
+ * \code
+ * NONSTRING
+ * static char const foo[3] = ".!?";
+ * \endcode
+ */
+#if __has_attribute(nonstring) || defined(__GNUC__)
```

Comment:
I doubt this attribute is available in every version of gcc. I think we should either drop `__GNUC__` or check its value (version).

Reply:
Fixed.

```diff
-#if __has_attribute(nonstring) || defined(__GNUC__)
+#if __has_attribute(nonstring)
```